packages/calcite-components/src/components/table-cell/table-cell.tsx

```diff
@@ -0,0 +1,204 @@
+
+/**
+ * @slot - A slot for adding content, usually text content.
+ */
+@Component({
+  tag: "calcite-table-cell",
+  styleUrl: "table-cell.scss",
+  shadow: true,
+  assetsDirs: ["assets"],
+})
+export class TableCell implements LocalizedComponent, LoadableComponent, T9nComponent {
+  //--------------------------------------------------------------------------
+  //
+  //  Properties
+  //
+  //--------------------------------------------------------------------------
+
+  /** Specifies the alignment of the component. */
+  @Prop({ reflect: true }) alignment: Alignment = "start";
+
+  /** Specifies the number of columns the component should span. */
+  @Prop({ reflect: true }) colSpan: number;
+
+  /** Specifies the number of rows the component should span. */
+  @Prop({ reflect: true }) rowSpan: number;
+
+  /** @internal */
+  @Prop() disabled: boolean;
+
+  /** @internal */
+  @Prop() numberCell: boolean;
+
+  /** @internal */
+  @Prop() parentRowIsSelected: boolean;
+
+  /** @internal */
+  @Prop() parentRowPositionLocalized: string;
+
+  /** @internal */
+  @Prop() parentRowType: RowType;
+
+  /** @internal */
+  @Prop() positionInRow: number;
+
+  /** @internal */
+  @Prop() scale: Scale = "m";
+
+  /** @internal */
+  @Prop() selectionCell: boolean;
+
+  /**
+   * Made into a prop for testing purposes only
+   *
+   * @internal
+   */
+  // eslint-disable-next-line @stencil-community/strict-mutable -- updated by t9n module
+  @Prop({ mutable: true }) messages: TableCellMessages;
+
+  /**
+   * Use this property to override individual strings used by the component.
+   */
+  // eslint-disable-next-line @stencil-community/strict-mutable -- updated by t9n module
+  @Prop({ mutable: true }) messageOverrides: Partial<TableCellMessages>;
+
+  @Watch("messageOverrides")
+  onMessagesChange(): void {
+    /* wired up by t9n util */
+  }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Private Properties
+  //
+  // --------------------------------------------------------------------------
+  @Element() el: HTMLCalciteTableCellElement;
+
+  /* Workaround for Safari https://bugs.webkit.org/show_bug.cgi?id=258430 https://bugs.webkit.org/show_bug.cgi?id=239478 */
+  /* q - should this be a state on a parent table and passed via internal prop? */
+  @State() isSafari: boolean;
+
+  @State() safariText = "";
+
+  @State() defaultMessages: TableCellMessages;
+
+  @State() effectiveLocale = "";
+
+  @Watch("effectiveLocale")
+  effectiveLocaleChange(): void {
+    updateMessages(this, this.effectiveLocale);
+  }
+
+  private tdEl: HTMLDivElement;
+
+  // --------------------------------------------------------------------------
+  //
+  //  Lifecycle
+  //
+  // --------------------------------------------------------------------------
+
+  async componentWillLoad(): Promise<void> {
+    setUpLoadableComponent(this);
+    await setUpMessages(this);
+    this.isSafari = /safari/i.test(getUserAgentString());
```

Comment:
Maybe this could be done once in a const outside of every `table cell` element instance when it is loaded?

Reply:
Yeah, probably best to determine once in Table and then pass to cell.

Reply:
Moved this to Table